xbmc/windowing/gbm/drm/DRMUtils.cpp

```diff
@@ -160,79 +157,166 @@ bool CDRMUtils::FindPreferredMode()
+          m_video_plane = video_plane.get();
+          goto success;
+        }
+        // if zpos is not supported, make sure gui plane id > video plane id
+        // this is how drm sorts when zpos is not available. Set crtc as the current crtc
+        else if (gui_plane->GetId() > video_plane->GetId())
+        {
+          m_crtc = m_crtcs[crtc_offset].get();
+          m_gui_plane = gui_plane.get();
+          m_video_plane = video_plane.get();
+          goto success;
+        }
+      }
+    }
+  }
 
-    auto videoPlane = std::ranges::find_if(m_planes,
-                                           [&i](auto& plane)
-                                           {
-                                             if (plane->GetPossibleCrtcs() & (1 << i))
-                                             {
-                                               return plane->SupportsFormat(DRM_FORMAT_NV12);
-                                             }
-                                             return false;
-                                           });
+  CLog::Log(
+      LOGERROR, "CDRMUtils::{} - Can not find a Video Plane plane with format {}, modifier {}",
+      __FUNCTION__, DRMHELPERS::FourCCToString(format), DRMHELPERS::ModifierToString(modifier));
+  return false;
 
-    uint32_t videoPlaneId{0};
+success:
+  CLog::Log(LOGINFO,
+            "CDRMUtils::{} - Switched GUI Plane to id:{}, video plane to id:{} on crtc id:{} video "
+            "for format:{}, modifier:{}",
+            __FUNCTION__, m_gui_plane->GetId(), m_video_plane->GetId(), m_crtc->GetId(),
+            DRMHELPERS::FourCCToString(format), DRMHELPERS::ModifierToString(modifier));
+  return true;
+}
 
-    if (videoPlane != m_planes.end())
-      videoPlaneId = videoPlane->get()->GetPlaneId();
+// determines the GUI rendering format and selects a plane+crtc for it without considering the future video plane
+bool CDRMUtils::InitGuiPlane(CEGLContextUtils* eglContext, EGLint renderableType)
+{
+  bool useCompression = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+      SETTING_VIDEOPLAYER_COMPRESSION);
 
-    auto guiPlane = std::ranges::find_if(
-        m_planes,
-        [&i, &videoPlaneId](auto& plane)
-        {
-          if (plane->GetPossibleCrtcs() & (1 << i))
-          {
-            return (plane->GetPlaneId() != videoPlaneId &&
-                    (videoPlaneId == 0 || plane->SupportsFormat(DRM_FORMAT_ARGB8888)) &&
-                    (plane->SupportsFormat(DRM_FORMAT_XRGB2101010) ||
-                     plane->SupportsFormat(DRM_FORMAT_XRGB8888)));
-          }
-          return false;
-        });
+  m_gui_plane = nullptr;
+  m_video_plane = nullptr;
+  m_crtc = nullptr;
 
-    if (videoPlane != m_planes.end() && guiPlane != m_planes.end())
+  // add more formats here in future
+  std::map<std::uint32_t, std::vector<uint32_t>> formats{
+      {DRM_FORMAT_ARGB8888, {DRM_FORMAT_ARGB8888, DRM_FORMAT_XRGB8888}},
```

Comment:
i guess this logic of falling to X variants of A formats will not work unless we activate ALPHA=0 in https://github.com/xbmc/xbmc/blob/60563b4cc382e51158b61d206e576afa41058a09/xbmc/utils/EGLUtils.cpp#L285-L297. I have manually set this to 0 then XRGB rendering works with transparency (actual transparency comes from kms planes). So KMS=AR24, EGL/GBM/BO=XR24

ie:
```
2025-10-31 16:28:41.759 T:913      info <general>: CDRMUtils::InitGuiPlane - Requested GUI plane is found with id: 122 and plane format AR24, egl format XR24 over crtc id: 89
2025-10-31 16:28:42.620 T:913      info <general>: CDRMUtils::DrmFbGetFromBo - BO FORMAT XR24
``